src/services/Config.php

```diff
@@ -915,6 +915,34 @@ private function _loadConfigSettings(string $category)
             }
         }
 
```

Comment:
I think the place this logic should happen is dynamically in `config/components/common.php`, when we're initializing the `db` component (much like we do for `dsn`).